MathLib/LinAlg/Eigen/EigenMapTools.h

```diff
@@ -0,0 +1,193 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MATHLIB_EIGENMAPTOOLS_H
+#define MATHLIB_EIGENMAPTOOLS_H
+
+#include <Eigen/Core>
+#include <cassert>
+#include <vector>
+
+namespace MathLib
+{
+/*! Creates an Eigen mapped matrix from the given data vector.
+ *
+ * \pre The passed \c data vector must have zero size.
+ * \post The \c data has size \c rows * \c cols.
+ *
+ * \note The data vector will have the same storage order (row or column major)
+ * as the requested matrix type.
+ */
+template <typename Matrix>
+Eigen::Map<Matrix> toZeroedMatrix(std::vector<double>& data,
```

Comment:
I'd like to see 'create' or 'new' or 'allocate' in the method name.
Currently no specific ideas, though.